client/web/src/stores/searchStack.ts

```diff
@@ -146,9 +147,20 @@ export function restorePreviousSession(): void {
     }
 }
 
-export function removeSearchStackEntry(entryToDelete: SearchStackEntry): void {
+export function removeFromSearchStack(idsToDelete: SearchStackEntryID | SearchStackEntryID[]): void {
     useSearchStackState.setState(currentState => {
-        const entries = currentState.entries.filter(entry => entry !== entryToDelete)
+        let entries: SearchStackEntry[]
+        if (Array.isArray(idsToDelete)) {
+            entries = [...currentState.entries]
+            for (const id of idsToDelete) {
+                entries.splice(
+                    entries.findIndex(entry => entry.id === id),
+                    1
+                )
+            }
+        } else {
+            entries = currentState.entries.filter(entry => entry.id !== idsToDelete)
+        }
```

Comment:
Why not use the same process for both single and multiple item removals? eg.


```suggestion
        if (!Array.isArray(idsToDelete)) {
            idsToDelete = [idsToDelete]
        }
        entries = [...currentState.entries]
        for (const id of idsToDelete) {
            entries.splice(
                entries.findIndex(entry => entry.id === id),
                1
            )
        }
```

Just seems a bit odd to me that one uses splice and the other uses filter.